packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -201,69 +200,128 @@ abstract class Scheduler extends BindingBase {
   /// Cancels the callback of the given [id].
   ///
   /// Removes the given callback from the list of frame callbacks. If a frame
-  /// has been requested does *not* cancel that request.
+  /// has been requested, this does not also cancel that request.
+  ///
+  /// Frame callbacks are registered using [scheduleFrameCallback] or
+  /// [addFrameCallback].
   void cancelFrameCallbackWithId(int id) {
     assert(id > 0);
     _transientCallbacks.remove(id);
     _removedIds.add(id);
   }
 
+  /// Asserts that there are no registered transient callbacks; if
+  /// there are, prints their locations and throws an exception.
+  ///
+  /// This is expected to be called at the end of tests (the
+  /// flutter_test framework does it automatically in normal cases).
+  ///
+  /// To invoke this method, call it, when you expect there to be no
+  /// transient callbacks registered, in an assert statement with a
+  /// message that you want printed when a transient callback is
+  /// registered, as follows:
```

Comment:
This runs on a bit too. A simple improvement might be to just leave out the "as follows".